minidump/src/minidump.rs

```diff
@@ -494,14 +501,21 @@ pub struct MinidumpException<'a> {
 }
 
 /// A list of memory regions included in a minidump.
+/// This is the underlying generic type for [MinidumpMemoryList] and [MinidumpMemory64List].
 #[derive(Debug)]
-pub struct MinidumpMemoryList<'a> {
-    /// The memory regions, in the order they were stored in the minidump.
-    regions: Vec<MinidumpMemory<'a>>,
+pub struct MinidumpMemoryListGeneric<'a, Descriptor> {
```

Comment:
I don't want to bikeshed too much, but having `Generic` is pretty uncommon in Rust type names. I don't have a great alternative, but here are a few suggestions to think about:

- `MinidumpMemory<T>`: this would require renaming the existing `MinidumpMemory`, we could use `MinidumpMemory32`?
- `MinidumpMemoryBase<T>`
- Just `Memory<T>`?